TeadsSampleApp/Controllers/Native/AppLovin/TableView/NativeAppLovinTableViewController.swift

```diff
@@ -0,0 +1,130 @@
+        let APPLOVIN_AD_UNIT_ID = pid
+        nativeAdLoader = MANativeAdLoader(adUnitIdentifier: APPLOVIN_AD_UNIT_ID)
+        
+        // Setting the modal parent view controller.
+        let teadsAdSettings = TeadsAdapterSettings { (settings) in
+            settings.enableDebug()
+            settings.pageUrl("https://www.teads.tv")
+        }
+        
+        nativeAdLoader.register(teadsAdSettings: teadsAdSettings)
+        
+        nativeAdLoader.nativeAdDelegate = self
+        nativeAdView = AppLovinNativeAdView.loadNib()
+        nativeAdView.bindViews(with: MANativeAdViewBinder { builder in
+            builder.titleLabelTag = 1
+            builder.advertiserLabelTag = 2
+            builder.bodyLabelTag = 3
+            builder.iconImageViewTag = 4
+            builder.optionsContentViewTag = 5
+            builder.mediaContentViewTag = 6
+            builder.callToActionButtonTag = 7
+        })
+        nativeAdLoader.loadAd(into: nativeAdView)
+    }
+    
+    func closeSlot(ad: TeadsAd) {
+        elements.removeAll { $0 == ad }
+        tableView.reloadData()
+    }
+}
+
+extension NativeAppLovinTableViewController: UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return elements.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        if indexPath.row == 0 {
+            let cell = tableView.dequeueReusableCell(withIdentifier: headerCell, for: indexPath)
+            return cell
+        } else if let nativeAdView = elements[indexPath.row] {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: teadsAdCellIndentifier, for: indexPath) as? NativeTableViewCell else {
+                return UITableViewCell()
+            }
+            nativeAdView.translatesAutoresizingMaskIntoConstraints = false
+            cell.nativeAdView.addSubview(nativeAdView)
+            nativeAdView.topAnchor.constraint(equalTo: cell.contentView.topAnchor).isActive = true
+            nativeAdView.bottomAnchor.constraint(equalTo: cell.contentView.bottomAnchor).isActive = true
+            nativeAdView.leadingAnchor.constraint(equalTo: cell.contentView.leadingAnchor).isActive = true
+            nativeAdView.trailingAnchor.constraint(equalTo: cell.contentView.trailingAnchor).isActive = true
+            
+            return cell
+        } else {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: fakeArticleCell, for: indexPath) as? FakeArticleNativeTableViewCell else {
+                return UITableViewCell()
+            }
+            cell.setMockValues()
+            return cell
+        }
+        
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        if elements[indexPath.row] != nil {
+            return 400
+        }
+        return 250
```

Comment:
elements should have the same height imo

Reply:
same as mopub integration